File: Selection_Sort/Selection_sort.rs
// Rust program for Selection Sort
use std::io;

fn main() {
    println!(" Enter number of elements to be entered in the array ");
    let mut size = String::new();
    
    io::stdin().read_line(&mut size).expect("failed to read input.");
    let size: usize = size.trim_end().parse().expect("invalid input");
    
    let mut vector: Vec<usize> = Vec::with_capacity(size as usize);
    
    println!("Enter elements of array ");
    let mut index = 0;
    
    // Enter values into vector
    while index < size {
        index += 1;
        // Note: Rust takes spaces as non-intergral value
        let mut temp_arr = String::new();
        io::stdin().read_line(&mut temp_arr).expect("failed to read input.");
        let temp_arr: usize = temp_arr.trim_end().parse().expect("invalid input");
        vector.push(temp_arr);
    }
    
    for index in 0..size - 1 {
        let mut minimum = index;
        let mut j = index + 1;
        while j < size {
            if vector[j] < vector[minimum] {
                minimum = j;
                j += 1;
            }
            // Swapping the smallest number in vector with the indexth location number
            let temp = vector[minimum];
            vector[minimum] = vector[index];
            vector[index] = temp;
        }
    }
    
    println!("Sorted array is ");
    for index in 0..size {
        println!("{:?}",vector[index]);
    }
}
/* 
TEST CASE

INPUT
 Enter number of elements to be entered in the array                                                                          
3                                                                                                                             
Enter elements of array                                                                                                       
3                                                                                                                             
2                                                                                                                             
1  

OUTPUT
Sorted array is                                                                                                               
1                                                                                                                             
2                                                                                                                             
3                                                                                                                     
*/

File: Selection_Sort/Selection_Sort.dart
//selection sort 

import 'dart:io';

void selection_sort (List<int> arr, int n)
{
    int temp, min_index;
    
    for (int i = 0 ; i  < n - 1 ; i++)
    {
        min_index = i;
        
        for (int j = i + 1 ; j < n ; j++)
            if ( arr[j] < arr[min_index] )
                 min_index = j;
            
        if (i != min_index)
        {
            temp = arr[i];
            arr[i] = arr[min_index];
            arr[min_index] = temp;
        }
    } 
}

main()
{
    print ("Enter the size of array ");
    int size = int.parse (stdin.readLineSync());
    
    List <int> array = List();
    
    for (int i = 0; i < size; i++)
    {
        print ("Enter $i element ");
        var ele = int.parse (stdin.readLineSync());
        array.add (ele);
    }
   
    selection_sort (array, size);
    
    print (array);

}

/*
Enter the size of array 
5
Enter 0 element 
50
Enter 1 element
40
Enter 2 element 
30
Enter 3 element
20
Enter 4 element 
10
Sorted array is
[10, 20, 30, 40, 50]
*/
Selection_Sort/Selection_Sort.dart
